hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-1.x-replication/src/test/java/com/google/cloud/bigtable/hbase1_x/replication/HbaseToCloudBigtableReplicationEndpointTest.java

```diff
@@ -450,34 +450,53 @@ public void testReplicationWithScope() throws IOException, InterruptedException
 
     Put put1 = new Put(ROW_KEY);
     put1.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
-    put1.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    put1.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER_2, 0, ROW_KEY);
     hbaseTable.put(put1);
 
     TestUtils.waitForReplication(
         () -> {
-          //  replicate Entries will be 1
           return TestReplicationEndpoint.replicatedEntries.get() >= 1;
         });
 
     Result cbtResult = cbtTable.get(new Get(ROW_KEY).setMaxVersions());
     Result hbaseResult = hbaseTable.get(new Get(ROW_KEY).setMaxVersions());
+
+    // make sure we have replicated cells
     Assert.assertFalse(cbtResult.isEmpty());
-    Assert.assertFalse(hbaseResult.isEmpty());
-    Assert.assertEquals(
-        "Number of cells , actual cells: " + hbaseResult.listCells(),
-        2,
-        2,
-        hbaseResult.listCells().size());
+    List<Cell> hbaseCells = hbaseResult.listCells();
+    List<Cell> cbtCells = cbtResult.listCells();
+
+    Assert.assertEquals("bigtable cells", 1, cbtCells.size());
+    // make sure that only CF1 is replicated
+    for (int i = 0; i < hbaseCells.size(); i++) {
+      // make sure CF1 is only replicated
+      if (CellUtil.cloneFamily(hbaseCells.get(i)) == CF1) {
+        assertEquals(hbaseCells.get(i), cbtCells.get(0));
+      }
+    }
+  }
 
-    Assert.assertEquals(
-        "Number of cells , actual cells: " + hbaseResult.listCells(),
-        1,
-        1,
-        cbtResult.listCells().size());
-    TestUtils.assertEquals(
-        hbaseResult,
-        cbtResult,
-        HConstants.REPLICATION_SCOPE_GLOBAL,
-        HConstants.REPLICATION_SCOPE_LOCAL);
+  @Test
+  public void testHBaseCBTTimestampTruncation() throws IOException, InterruptedException {
+    Put put = new Put(TestUtils.ROW_KEY);
+    byte[] val = Bytes.toBytes(1);
+    put.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, Long.MAX_VALUE - 1, val);
+    put.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, Long.MAX_VALUE - 1000, val);
+    hbaseTable.put(put);
+
+    TestUtils.waitForReplication(
+        () -> {
+          // 1put
+          return TestReplicationEndpoint.replicatedEntries.get() >= 1;
+        });
+
+    List<Cell> hbaseCells = hbaseTable.get(new Get(TestUtils.ROW_KEY).setMaxVersions()).listCells();
+    List<Cell> bigtableCells =
+        cbtTable.get(new Get(TestUtils.ROW_KEY).setMaxVersions()).listCells();
+    Assert.assertEquals("bigtable cells", 1, bigtableCells.size());
```

Comment:
We can improve message here. You should also add a comment explaining that timestamp will get truncated and value will be overwritten at BIGTABLE_MAX_TIMESTAMP.